<launch>
	<arg name="drones" default="[1 2]"/>
  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>
  <arg name="drone_id" default="1"/>
  <arg name="trajectory_frame" default="$(arg uav_name)/gps_origin"/>
  <arg name="run_rviz" default="true" />
  <arg name="pcl_map_filename" default="$(find formation_church_planning)/maps/pcd_files/power_tower_danube.pcd" />
  <arg name="config_name" default="two_drones.yaml"/>

    <node pkg="optimal_control_interface" name="solver" type="optimal_control_interface_node" output="screen" ns="$(arg uav_name)">
      <param name="drone_id" value="$(arg drone_id)"/>
      <param name="uav_name" value="$(arg uav_name)"/>
      <param name="solver_rate" value="0.5"/> <!-- Hz -->
      <param name="trajectory_frame" value="$(arg trajectory_frame)"/> <!-- Hz -->
      <rosparam param="no_fly_zone" subst_value="true">[0 0]</rosparam>
      <!-- remapping of topics -->
      <remap from="~odometry_in" to="odometry/odom_main" />
      <remap from="~diagnostics" to="formation_church_planning/diagnostics" />
      <remap from="~desired_trajectory" to="control_manager/trajectory_reference" />
      <remap from="~planned_trajectory" to="formation_church_planning/planned_trajectory" />
      <remap from="~mrs_status" to="formation_church_planning/status" />
      <remap from="~toggle_state" to="formation_church_planning/toggle_state" />
      <remap from="~target_topic" to="/gazebo/dynamic_model/jeff_electrician/odometry" />
      <!-- <remap from="~target_topic" to="/$(arg uav_name)/balloon_filter/chosen_out" /> -->
      <remap from="~target_odometry_out" to="/$(arg uav_name)/formation_church_planning/target_odometry" />

      <rosparam file="$(find safe_corridor_generator)/config/params.yaml" />
      <rosparam file="$(find formation_church_planning)/config/$(arg config_name)" />

      

      <param name="pcl_filepath" value="$(arg pcl_map_filename)" />
      <param name="world_frame" value="$(arg uav_name)/gps_origin" />
      <param name="drones_ids" value="[10 6 7]" /> 

      <remap from="~path_out" to="~path_out" />
      <remap from="~path_in" to="~path_in" />
      <remap from="~pcl_map_out" to="~pcl_map_out" />
      <remap from="~polyhedrons_out" to="~polyhedrons_out" />
      <remap from="~ellipsoids_out" to="~ellipsoids_out" />

    </node>


</launch>

